server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesIndexAction.java

```diff
@@ -86,6 +87,20 @@ protected FieldCapabilitiesIndexResponse shardOperation(final FieldCapabilitiesI
                 if (indicesService.isMetaDataField(field) || fieldPredicate.test(ft.name())) {
                     FieldCapabilities fieldCap = new FieldCapabilities(field, ft.typeName(), ft.isSearchable(), ft.isAggregatable());
                     responseMap.put(field, fieldCap);
+                } else {
+                    continue;
+                }
+                // add nested and object fields
+                int dotIndex = ft.name().indexOf('.');
+                while (dotIndex > -1) {
+                    String parentField = ft.name().substring(0, dotIndex);
+                    ObjectMapper mapper = mapperService.getObjectMapper(parentField);
+                    if (mapper != null) {
```

Comment:
I was wondering whether this condition may ever be false, maybe mention sub fields in a comment? Should we also break as soon as `responseMap.containsKey(parentField)` is true?

Reply:
> Should we also break as soon as responseMap.containsKey(parentField) is true?

If we start from the leaf, yes, I pushed 
https://github.com/elastic/elasticsearch/pull/33803/commits/235df36377ff4c2e171c451bc748097f0ad9b866 to reverse the iteration order.